### PR TITLE
achievements code

### PR DESCRIPTION
**Pull request by BaegEhl**
this *should* have the code for achievements
adds: achievement script with trackers for credits gained and spent, enemy shellcores killed, sectors explored, parts collected and unique parts collected
needs: GUI (idk how to do that), top speed tracker, probably needs to be refactored, might not actually work either

### CODE REVIEW

File: Assets/Scripts/Game Object Definitions/Entity Definitions/ShellCore.cs [hunk 1]
```diff
@@ -16,8 +16,8 @@ public class ShellCore : AirCraft, IHarvester, IOwner {
     public int intrinsicCommandLimit;
     public List<IOwnable> unitsCommanding = new List<IOwnable>();
 
-    private TractorBeam tractor;
-
+    private TractorBeam tractor; 
+    public PlayerCore player;
```
Comment (rudderbucky) [inline Assets/Scripts/Game Object Definitions/Entity Definitions/ShellCore.cs:20]: Unnecessary

File: Assets/Scripts/HUD Scripts/AchievementScript.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    public void updateParts(){
+        partsCollected = save.partInventory.Count;
+    }
+    public void updateUniqueParts(){
+        uniqueParts = save.partsObtained.Count;
+    }
+    public void updateCreditsGain(){
+        creditGain = save.creditsGained;
+    }
+    public void updateCreditsSpend(){
```
Comment (rudderbucky) [inline Assets/Scripts/HUD Scripts/AchievementScript.cs:27]: Typo, should be updateCreditsSpent

Comment (BaegEhl) [reply Assets/Scripts/HUD Scripts/AchievementScript.cs]: fixed

**Comment by rudderbucky** [comment]
Achievements is gonna require more work than this on the front end. I think this is a good start but IDK if I can give the time for achievements right now. Perhaps try implementing it in the status menu?

**Comment by BaegEhl** [comment]
statistics seem to work and achievement canvas is accessible but it doesn't work and I don't have any idea how to make it work

**Pull request closed without merging**